pandas/core/strings.py

```diff
@@ -1962,11 +1934,269 @@ def cons_row(x):
                 cons = self._orig._constructor
                 return cons(result, name=name, index=index)
 
-    @copy(str_cat)
-    def cat(self, others=None, sep=None, na_rep=None):
-        data = self._orig if self._is_categorical else self._data
-        result = str_cat(data, others=others, sep=sep, na_rep=na_rep)
-        return self._wrap_result(result, use_codes=(not self._is_categorical))
+    def _str_cat_los(self, input, ignore_index=False):
+        """
+        Auxiliary function for :meth:`str.cat`. Turn potentially mixed input
+        into list of Series (elements without an index must match the length of
+        the calling Series/Index).
+
+        Parameters
+        ----------
+        input : Series, DataFrame, np.ndarrary, list-like or list-like of those
```

Comment:
1D ndarray, right? we accept a list-like of DataFrames? really?

Reply:
This is debatable, I agree. I preferred the recursive structure of the (renamed) ``_get_series_list``, and I didn't start restricting certain cases from that. The docs and tests also contain this case.